src/commands.ts

```diff
@@ -147,78 +147,33 @@ export class Commands {
     // a host label.
     const label = typeof args[2] === "undefined" ? toSafeHost(url) : args[2]
 
-    // Use a temporary client to avoid messing with the global one while trying
-    // to log in.
-    const restClient = await makeCoderSdk(url, undefined, this.storage)
-
-    let user: User | undefined
-    let token: string | undefined = args[1]
-    if (!token) {
-      const opened = await vscode.env.openExternal(vscode.Uri.parse(`${url}/cli-auth`))
-      if (!opened) {
-        vscode.window.showWarningMessage("You must accept the URL prompt to generate an API key.")
-        return
-      }
-
-      token = await vscode.window.showInputBox({
-        title: "Coder API Key",
-        password: true,
-        placeHolder: "Copy your API key from the opened browser page.",
-        value: await this.storage.getSessionToken(),
-        ignoreFocusOut: true,
-        validateInput: async (value) => {
-          restClient.setSessionToken(value)
-          try {
-            user = await restClient.getAuthenticatedUser()
-            if (!user) {
-              throw new Error("Failed to get authenticated user")
-            }
-          } catch (err) {
-            // For certificate errors show both a notification and add to the
-            // text under the input box, since users sometimes miss the
-            // notification.
-            if (err instanceof CertificateError) {
-              err.showNotification()
-
-              return {
-                message: err.x509Err || err.message,
-                severity: vscode.InputBoxValidationSeverity.Error,
-              }
-            }
-            // This could be something like the header command erroring or an
-            // invalid session token.
-            const message = getErrorMessage(err, "no response from the server")
-            return {
-              message: "Failed to authenticate: " + message,
-              severity: vscode.InputBoxValidationSeverity.Error,
-            }
-          }
-        },
-      })
-    }
-    if (!token || !user) {
-      return
+    // Try to get a token from the user, if we need one, and their user.
+    const res = await this.maybeAskToken(url, args[1])
```

Comment:
Nonblocking Nit: Validating the args passed in and giving them descriptive variable names should happen in one section at the top.

Reply:
Definitely agree, this was just old code moved and indented but it is a good opportunity to update.

Reply:
Wait no I did actually introduce this??